src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/GeneratedComInterfaceAttributeAnalyzer.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+#nullable enable
+namespace Microsoft.Interop
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class GeneratedComInterfaceAttributeAnalyzer : DiagnosticAnalyzer
+    {
+        private const string _generatedComInterfaceAttributeName = "GeneratedComInterfaceAttribute";
+        private const string _generatedComInterfaceAttributeNamespace = "System.Runtime.InteropServices.Marshalling";
+#if DEBUG
+        private const string _generatedComInterfaceAttributeAssemblyName = "Microsoft.Interop.Ancillary";
+#else
+        private const string _generatedComInterfaceAttributeAssemblyName = "System.Runtime.InteropServices";
+#endif
```

Comment:
We can just match the attribute based on name. We don't need to match on the symbol.